internal/impl/sftp/output.go

```diff
@@ -57,7 +70,13 @@ func sftpOutputSpec() *service.ConfigSpec {
 				Examples("lines", "delim:\t", "delim:foobar").
 				Default("all-bytes"),
 			service.NewObjectField(soFieldCredentials, credentialsFields()...).
-				Description("The credentials to use to log into the target server."),
+				Description("The credentials to use to log into the target server.").
+				LintRule(`
+root = match {
+  this.exists("host_public_key") && this.exists("host_public_key_file") => "both host_public_key and host_public_key_file can't be set simultaneously"
+  this.exists("private_key") && this.exists("private_key_file") => "both private_key and private_key_file can't be set simultaneously"
+}`,
+				),
```

Comment:
Should be also applied to input and tested.

Reply:
I need to find an example of how we can add tests for lint rules. Will do some digging to figure it out.